# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1262849501 -3600
#      Thu Jan 07 08:31:41 2010 +0100
# Node ID 897e62651c0adc8bc682c30a406b5ef69806e4e1
# Parent  c670c8be7f938bb75599f0feda0496250d231b08
correctly handle interrupts in pause()

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2010-01-07  Jaroslav Hajek  <highegg@gmail.com>
+
+	* utils.cc (octave_sleep (double)): Add OCTAVE_QUIT.
+
 2010-01-06  Jaroslav Hajek  <highegg@gmail.com>
 
 	* unwind_protect.h, unwind_protect.cc (unwind_protect): Rewrite.
 	Remove obsolete macros. Use a simple linked list.
 
 	* debug.cc (Fdbwhere): Update to new unwind_protect style.
 	* dynamic-ld.cc (octave_dynamic_loader::do_load_oct,
 	  octave_dynamic_loader::do_load_mex): Ditto.
diff --git a/src/utils.cc b/src/utils.cc
--- a/src/utils.cc
+++ b/src/utils.cc
@@ -1339,16 +1339,18 @@ octave_sleep (double seconds)
 
       unsigned int sec
 	= (t > UINT_MAX) ? UINT_MAX : static_cast<unsigned int> (t);
 
       // Versions of these functions that accept unsigned int args are
       // defined in cutils.c.
       octave_sleep (sec);
       octave_usleep (usec);
+
+      OCTAVE_QUIT;
     }
 }
 
 DEFUN (isindex, args, ,
   "-*- texinfo -*-\n\
 @deftypefn {Built-in Function} {} isindex (@var{ind}, @var{n})\n\
 Returns true if @var{ind} is a valid index.  Valid indices can be\n\
 either positive integers (though possibly real data), or logical arrays.\n\
